OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -406,5 +404,10 @@ public bool HasFogVisibility()
 		{
 			return fogVisibilities.Value.Any(f => f.HasFogVisibility(self.Owner));
 		}
+
+		public bool Contains(MPos uv)
+		{
+			return explored.Contains(uv);
```

Comment:
This isn't too complicated, but still got me confused. Please add a comment why you're checking `explored`.